gtsfm/evaluation/metrics_report.py

```diff
@@ -125,22 +158,99 @@ def get_figures_for_metrics(metrics_group: GtsfmMetricsGroup) -> Tuple[str, str]
             if metrics.SUMMARY_KEY not in value:
                 raise ValueError(f"Metric {metric_name} does not contain a summary.")
-            # Add a scalar metric for mean of 1D distributions.
-            scalar_metrics["mean_" + metric_name] = value[metrics.SUMMARY_KEY]["mean"]
+            # Add a scalar metric for median of 1D distributions.
+            scalar_metrics["median_" + metric_name] = value[metrics.SUMMARY_KEY]["median"]
         else:
             scalar_metrics[metric_name] = value
     table = create_table_for_scalar_metrics(scalar_metrics)
     plots_fig = create_plots_for_distributions(metrics_dict)
     return table, plots_fig
 
 
+def add_summary(metric_name, metric_value, scalar_metrics):
+    # Metrics with a dict representation must contain a summary.
+    if metrics.SUMMARY_KEY not in metric_value:
+        raise ValueError(f"Metric {metric_name} does not contain a summary.")
+    # Add a scalar metric for median of 1D distributions.
+    median_nan = metric_value[metrics.SUMMARY_KEY]["median"] != metric_value[metrics.SUMMARY_KEY][
+        "median"]
+    if median_nan:
+        scalar_metrics["median_" + metric_name].append("")
+    else:
+        scalar_metrics["median_" + metric_name].append(metric_value[metrics.SUMMARY_KEY]["median"])
+
+def add_metric(gtsfm_metric_name, gtsfm_metric_value, scalar_metrics, other_pipelines_metric_dicts):
+    """
+    Args:
+        gtsfm_metric_name: Use GTSfM metric name for adding metrics (other pipelines share metric names)
+        gtsfm_metric_value: GTSfM metric value
+        scalar_metrics: Dictionary with metric names as keys and a list of metric values for each pipeline
+        other_pipelines_metric_dicts: Dictionaries of metric name, metric value key-value pairs for each other pipeline
+
+    """
+    other_pipelines_metric_values = []
+    for other_pipeline_metric_dict in other_pipelines_metric_dicts:
+        if gtsfm_metric_name in other_pipeline_metric_dict:
+            other_pipelines_metric_values.append(other_pipeline_metric_dict[gtsfm_metric_name])
+        else:
+            other_pipelines_metric_values.append("")
+
+    if isinstance(gtsfm_metric_value, dict):
+        add_summary(gtsfm_metric_name, gtsfm_metric_value, scalar_metrics)
+        for other_pipeline_metric_value in other_pipelines_metric_values:
+            if isinstance(other_pipeline_metric_value, dict):
+                add_summary(gtsfm_metric_name, other_pipeline_metric_value, scalar_metrics)
+            else:
+                other_pipeline_metric_value = {'summary': {'median': ""}}
+                add_summary(gtsfm_metric_name, other_pipeline_metric_value, scalar_metrics)
+    else:
+        scalar_metrics[gtsfm_metric_name].append(gtsfm_metric_value)
+        for other_pipeline_metric_value in other_pipelines_metric_values:
+            scalar_metrics[gtsfm_metric_name].append(other_pipeline_metric_value)
```

Comment:
as my other comment above says, I think we want to be functional here, so could we return a key, value maybe?

Reply:
Sure thing, now using key, value.